src/modules/roc_core/target_posix/roc_core/secure_random.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2020 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//! @file roc_core/target_posix/roc_core/secure_random.h
+//! @brief Helpers to work with CSPRNG.
+
+#ifndef ROC_CORE_SECURE_RANDOM_H_
+#define ROC_CORE_SECURE_RANDOM_H_
+
+#include "roc_core/stddefs.h"
+
+namespace roc {
+namespace core {
+
+//! Get a random integer from a cryptographically secure pseudorandom number
+//! generator(CSPRNG).
+//! @returns secure random value in range [from; to].
+bool secure_random(uint32_t from, uint32_t to, uint32_t& result);
```

Comment:
We should improve the comment a bit:

1. `@returns` section is not quite correct. The function *returns* success status.
2. We should mention that it may block, it's important.
3. We should mention that it may fallback to a regular CSPRNG.

(Actually I already asked for this earlier in comments).

I suggest something like:

```
//! Get a random integer from a cryptographically secure pseudo-random number
//! generator (CSPRNG).
//! @remarks
//!   Writes a random value in range [from; to] to @p result.
//!   May block for a while.
//!   Falls back to a regular PRNG when using libuv versions before 1.33.0.
//! @returns
//!   true on success or false if accessing CSPRNG failed.
```

Reply:
thanks 👍